packages/dashboard/components/consent/pages/index.tsx

```diff
@@ -23,13 +25,58 @@ const CenteredContainer = ({ children }: { children: React.ReactNode }) => (
 );
 
 export default function IndexView() {
-  const [sessionSigs, setSessionSigs] = useState<SessionSigs>();
-  const [agentPKP, setAgentPKP] = useState<IRelayPKP>();
-  const [sessionLoading, setSessionLoading] = useState<boolean>(false);
-  const [sessionError, setSessionError] = useState<Error>();
+  // ------ STATE AND HOOKS ------
   const { showError } = useErrorPopup();
   const { updateAuthInfo } = useSetAuthInfo();
+  const clearAuthInfo = useClearAuthInfo();
 
+  // Shared state for session sigs and agent PKP
+  const [sessionSigs, setSessionSigs] = useState<SessionSigs>();
+  const [agentPKP, setAgentPKP] = useState<IRelayPKP>();
+  const [sessionError, setSessionError] = useState<Error>();
+  
+  // State for loading messages
+  const [loadingMessage, setLoadingMessage] = useState<string>('');
+  const [isTransitioning, setIsTransitioning] = useState(false);
+  
+  // ------ EXISTING SESSION HANDLING ------
+  
+  // Check for existing auth info
+  const { 
+    authInfo, 
+    sessionSigs: validatedSessionSigs, 
+    isProcessing,
+    error: readError
+  } = useReadAuthInfo();
+  
+  // State to show existing account option
+  const [showExistingAccount, setShowExistingAccount] = useState(false);
+  
+  // Check for existing auth info once the validation process is complete
+  useEffect(() => {
+    if (isProcessing) return;
+    if (validatedSessionSigs) {
+      setShowExistingAccount(true);
+    }
+  }, [validatedSessionSigs, isProcessing]);
+  
+  // Handle using existing account
+  const handleUseExistingAccount = () => {
+    if (!validatedSessionSigs || !authInfo) return;
+    
+    // Use the existing session and auth info
+    setSessionSigs(validatedSessionSigs);
+    setuserPKP(authInfo.userPKP);
+    setAgentPKP(authInfo.agentPKP);
+    setShowExistingAccount(false);
+  };
```

Comment:
🔪 🔪 there's no need to set the same data into multiple duplicate variable/state holders here. All we need to do is set the flag for `showExistingAccount` -- kill all 3 of these setters and reference the values directly from the object exposed from the `useReadAuthInfo()` hook